server/src/main/java/com/google/musicanalysis/site/OAuthLoginServlet.java

```diff
@@ -0,0 +1,41 @@
+package com.google.musicanalysis.site;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.URLEncoder;
+import java.security.SecureRandom;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/api/oauth/login")
+public class OAuthLoginServlet extends HttpServlet {
```

Comment:
**This servlet requests authorization; the user logs in and authorizes access** 

[see step 1 for more info](https://developer.spotify.com/documentation/general/guides/authorization-guide/)